library/L2_HAL/sensors/environment/temperature/test/tmp102_test.cpp

```diff
@@ -0,0 +1,122 @@
+#include "L1_Peripheral/system_controller.hpp"
+#include "L2_HAL/sensors/environment/temperature/tmp102.hpp"
+#include "L4_Testing/testing_frameworks.hpp"
+
+namespace sjsu
+{
+EMIT_ALL_METHODS(Tmp102);
+
+TEST_CASE("Testing Tmp102 Temperature Sensor", "[tmp102]")
+{
+  constexpr uint8_t kDeviceAddress = Tmp102::DeviceAddress::kGround;
+  Mock<I2c> mock_i2c;
+  Tmp102 temperature_sensor(mock_i2c.get(), kDeviceAddress);
+
+  When(Method(mock_i2c, Initialize)).AlwaysReturn(Status::kSuccess);
+
+  SECTION("Initialize")
+  {
+    const Status kInitializationStatus = temperature_sensor.Initialize();
+    CHECK(kInitializationStatus == Status::kSuccess);
```

Comment:
It maybe useful to check that your driver actually passes a non-success status back. So if init did fail we have a check for that here.